### ISSUE TITLE
ndigits(x::Int128, 256) loops for x >= 2^64

### ISSUE THREAD

**Issue by KlausC** [JuliaLang/julia#19367]
```julia
Version 0.5.0 (2016-09-19 18:14 UTC)
x86_64-redhat-linux
julia> x = Int128(2)^64
18446744073709551616
julia> ndigits(x, 256)                         <============ this call loops forever !!!
^C^C^C^C^C^C^C^CWARNING: Force throwing a SIGINT
^CERROR: InterruptException:
 in ndigits0z(::UInt128, ::Int64) at ./intfuncs.jl:245
 in ndigits at ./intfuncs.jl:256 [inlined]
 in ndigits(::Int128, ::Int64) at ./intfuncs.jl:259
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:64
 in macro exjulia> x = 18446744073709551615
18446744073709551615

julia> ndigits(x, 256)
8
pansion at ./REPL.jl:95 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at ./event.jl:68

julia> x = 18446744073709551615;
julia> ndigits(x, 256)
8
```

**Comment by simonbyrne** [comment]
The problem seems to be that [this test](https://github.com/JuliaLang/julia/blob/d2a19d880a41838cac28ea997d2ceee9b3eb42a7/base/intfuncs.jl#L337) presumes that the base `b` is less than 31 (otherwise the multiplication can overflow).

Interestingly, this bug has been around for a long time: at least since 2012.

**Comment by simonbyrne** [comment]
fixed by #19373
